homeassistant/components/apsystems/switch.py

```diff
@@ -0,0 +1,53 @@
+"""The power switch which can be toggled via the APsystems local API integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ApSystemsConfigEntry, ApSystemsData
+from .entity import ApSystemsEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ApSystemsConfigEntry,
+    add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the switch platform."""
+
+    add_entities([ApSystemsPowerSwitch(config_entry.runtime_data)])
+
+
+class ApSystemsPowerSwitch(ApSystemsEntity, SwitchEntity):
+    """The switch class for APSystems switches."""
+
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _attr_translation_key = "power_output"
+
+    def __init__(self, data: ApSystemsData) -> None:
+        """Initialize the switch."""
+        super().__init__(data)
+        self._api = data.coordinator.api
+        self._attr_unique_id = f"{data.device_id}_power_output"
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return super().available and bool(self._api.get_device_power_status())
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return state of the switch."""
+        return bool(self._api.get_device_power_status())
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the switch on."""
+        await self._api.set_device_power_status(0)
```

Comment:
Idk, but running `await self.async_update()` may be needed here, but I'm not sure, another one will have to verify

Reply:
I have tested it locally in connection with my inverter. And it looks like it works without an additional `await self.async_update()` at this point.

Reply:
Yes, the other integrations also don't call the update function